addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -90,3 +90,17 @@ def test_notify_reply_to_computation_mc(self):
                     "%s %s" % (self.user_employee_c2.company_id.name, test_record.name),
                     "%s@%s" % (self.alias_catchall, self.alias_domain)))
             )
+
+        # Test3: get company from record (company_id field)
+        new_company = self.env['res.company'].sudo().create({'name': 'ELIT'})
+        test_record = self.env['mail.test.multi.company'].create({
```

Comment:
What about creating 2 records in the new company (len(company_id) will be 1) and run the call '_notify_get_reply_to' as multi on those two records, just to test a bit the batch mode ?